test/codegen/end_to_end_test.dart

```diff
@@ -121,6 +121,10 @@ void main(List<String> args) {
     runQuery('K', new QueryOptions(
         includeInherited: true, withAnnotations: [lib.getType('AnnotC')]));
 
+    runQuery('J', new QueryOptions(includeMethods: true));
+    runQuery('J2', new QueryOptions(
```

Comment:
Made an attempt. Would feel a bit more comfortable tackling this and then moving on to that request.